vpc/service/assign_addresses_v3.go

```diff
@@ -1134,85 +1137,30 @@ func assignArbitraryIPv6AddressV3(ctx context.Context, tx *sql.Tx, branchENI *ec
 	ctx, span := trace.StartSpan(ctx, "assignArbitraryIPv6AddressV3")
 	defer span.End()
 
-	usedIPAddresses := sets.NewString()
-	rows, err := tx.QueryContext(ctx, "SELECT ipv6addr FROM assignments WHERE ipv6addr IS NOT NULL AND branch_eni_association = $1", ass.branch.associationID)
+	row := tx.QueryRowContext(ctx, "UPDATE subnet_usable_prefix SET last_assigned = last_assigned + 1 WHERE branch_eni_id = $1 RETURNING last_assigned, prefix", ass.branch.intid)
+	var lastAssigned int64
+	var prefix string
+	err := row.Scan(&lastAssigned, &prefix)
 	if err != nil {
-		err = errors.Wrap(err, "Cannot query ipv6addrs already assigned to interface")
-		span.SetStatus(traceStatusFromError(err))
-		return nil, err
-	}
-	for rows.Next() {
-		var address string
-		err = rows.Scan(&address)
-		if err != nil {
-			err = errors.Wrap(err, "Cannot scan rows")
-			span.SetStatus(traceStatusFromError(err))
-			return nil, err
-		}
-		usedIPAddresses.Insert(address)
-	}
-
-	allInterfaceIPAddresses := sets.NewString()
-	for idx := range branchENI.Ipv6Addresses {
-		allInterfaceIPAddresses.Insert(aws.StringValue(branchENI.Ipv6Addresses[idx].Ipv6Address))
-	}
-
-	if l := usedIPAddresses.Len(); l >= maxIPAddresses {
-		err = status.Errorf(codes.FailedPrecondition, "%d IPv6 addresses already in-use", l)
-		span.SetStatus(traceStatusFromError(err))
-		return nil, err
-	}
-
-	logger.G(ctx).WithField("usedIPAddresses", usedIPAddresses.List()).WithField("allInterfaceIPAddresses", allInterfaceIPAddresses.List()).Debug("Trying to assign IPv6 Address")
-	unusedIPAddresses := allInterfaceIPAddresses.Difference(usedIPAddresses)
-
-	if l := unusedIPAddresses.Len(); l > 0 {
-		unusedIPv6AddressesList := unusedIPAddresses.List()
-
-		row := tx.QueryRowContext(ctx, "SELECT ip_address FROM ip_last_used_v3 WHERE ip_address = any($1::inet[]) AND vpc_id = $2 ORDER BY last_seen ASC LIMIT 1", pq.Array(unusedIPv6AddressesList), aws.StringValue(branchENI.VpcId))
-		var ipAddress string
-		err = row.Scan(&ipAddress)
-		if err == sql.ErrNoRows {
-			// Effectively choose a random one.
-			ipAddress = unusedIPv6AddressesList[rand.Intn(l)] // nolint: gosec
-		} else if err != nil {
-			err = errors.Wrap(err, "Could not fetch utilized IPv6 addresses from the database")
-			tracehelpers.SetStatus(err, span)
-			return nil, err
-		}
-
-		return &vpcapi.UsableAddress{
-			Address: &vpcapi.Address{
-				Address: ipAddress,
-			},
-			PrefixLength: uint32(128),
-		}, nil
-	}
-
-	ipsToAssign := maxIPAddresses - allInterfaceIPAddresses.Len()
-	if ipsToAssign <= 0 {
-		err = status.Errorf(codes.FailedPrecondition, "%d IPv4 addresses already assigned to interface", allInterfaceIPAddresses.Len())
-		span.SetStatus(traceStatusFromError(err))
+		err = fmt.Errorf("Cannot query / scan row from subnet_usable_prefix: %w", err)
```

Comment:
I'm going to need an in-person walkthrough (maybe with @MyUmmaGumma too?) on this.

In general, this is cool, because it simplifies how we assign at all (just bump the int).

But I don't understand how we are going to get from where we are today, to here?
How can we be sure we will have subnets delegated in all the places where we know we are going to want ips?